ddtrace/tracer/tracer_test.go

```diff
@@ -1212,3 +1215,23 @@ func BenchmarkTracerStackFrames(b *testing.B) {
 		span.Finish(StackFrames(64, 0))
 	}
 }
+
+// Make sure the tracer flushes when the payload gets too big. (see #569)
+func TestPayloadSizeFlush(t *testing.T) {
```

Comment:
It seems the test "makes sure" as a side-effect, rather than directly observing that after `payloadSizeLimit` is exceeded, a flush is sent to the worker.